### ISSUE TITLE
Re-using LePhare CFHTLenS photo-zs?

### ISSUE THREAD

**Issue by drphilmarshall** [eduardrusu/zMstarPDF#5]
Is it possible to reconstruct the joint PDF Pr(z,Mstar | photometry) from the CFHTLenS outputs? In an ideal world we'd be provided with not only Pr(z | photometry) but also Pr(Mstar | z,photometry) (perhaps on a 2D grid), such that the product gives us what we need. I think this is a good question for the producers of the  catalogs: let's check their papers for the answer, and then contact them if we can't find it. 

Alternatively, we have some compressed information (Mstar estimates with confidence intervals, plus a tabulated Pr(z|photometry)) that could perhaps be interpreted as above, to give a useful approximation. I'm not sure if this approximation will be good enough for our purposes: it _might_ be. We must test it, if we have to go this route.

**Comment by drphilmarshall** [comment]
BTW I ran this plan past Eli Rykoff (of redMaPPer fame) and he was not too optimistic. Admittedly, he worries about photo-z accuracy to a greater extent than we do, but still, his opinion was that we should be running `stellarpops` on the CFHTLS galaxies as well as the H0LiCOW ones, for best results.

**Comment by eduardrusu** [comment]
- Ideally we need to run Stellarpops (Matt's code) on the lens fields and CFHTLens in order to get the full PDF Pr(z,Mstar | photometry); may be too time consuming to use on the whole CFHTLens, therefore likely need to limit to a subsample
- While getting the ful P(Mstar) for each object in CFHTLens is computationally prohibitive, getting just a rudimentary PDF (best-fit and confidence levels) should work. As Phil asked, is that enough?
- In order to test for systematics, we want to separately use EAZY, BPZ (for redshift), FAST (for stellar masses), Lephare (redshift and stellar masses). But these are not ideal when it comes to stellar masses. FAST uses as input a "rudimentary" (compressed information) PDF (best-fit value with 68,95,99 limits), and also outputs a rudimentary PDF for Mstar; it also uses a different set of template from those used to compute P(z). I need to check if it is possible to get it to use the same templates. On the other hand, to get the PDF for Mstar from Lephare I would need to sample from its P(z), fix the redshift, and rerun many times, which is computationally prohibitive for CFHTLens.
- What happens if we don't use PDF Pr(z,Mstar | photometry) at all, but just the best-fit values (usually the peaks of the distributions), as long as we use the same code for both the lens fields and the calibration fields? I did this in the histograms ("orig" and "samp") and there are vast differences (large offsets) compared to using the rudimentary PDFs. But need to remember that these are not the final PDFs, and for example I used different codes for the lens fields compared to those used for the CFHTLens catalogue values. 
- Keep in mind that when using different codes on CFHTLens, we are likely not doing as good a job as was done in estimating the P(z) given in the CFHTLens catalogues, because they were very careful to calibrate against spectroscopic samples. 
- CFHTLens: for redshift and stellar masses, they used BPZ to compute P(z) (tabulated values provided), but they used a fixed z (the most probable value) in computing stellar masses separately, with Lephare.

**Comment by eduardrusu** [comment]
I think I can close this. The method we used in Rusu et al 2017 was to considered the CFHTLenS-reported _z_ and Mstar (actually recomputed Mstar for the Millennium cosmology), whereas for the HE0435 field we used the best-fit _z_ and Mstar of each galaxy. In addition, for the HE0435 field we sampled from the Pr(z,Mstar | photometry) of each galaxy, and we took into account the resulting scatter when computing the ovedensities.
